### ISSUE TITLE
Handle resampling edge cases before returning invalid splits

### ISSUE THREAD

**Issue by FahadEbrahim** [FahadEbrahim/matheel#204]
## Summary
Handle resampling edge cases for scalar counts, impossible stratification, and empty bootstrap tests.

## Category
Evaluation methodology

## Motivation
Several resampling edge cases currently fail late or produce splits that are invalid for downstream evaluation. NumPy integer scalar counts are treated as sequences and crash. Stratified single splits can silently place the only member of a class entirely outside the training set. Bootstrap resamples for very small inputs can produce empty out-of-bag test sets that evaluation rejects.

## Proposed behavior
Normalize scalar counts more broadly and validate split feasibility before returning `DataSplit` objects that downstream evaluation cannot use.

## Scope and non-goals
In scope:
- Accept integer-like scalar counts such as `numpy.int64`.
- Warn or raise when stratified single split cannot preserve required classes.
- Decide and document expected bootstrap behavior when out-of-bag test sets are empty.
- Add focused regression tests.

Out of scope:
- Replacing the existing resampling API.
- Adding external resampling dependencies.

## Acceptance criteria
- `kfold_splits(np.int64(10), n_splits=3)` works or fails with a clear supported-input error.
- Stratified single split does not silently return a training set missing a class when preservation is impossible.
- Bootstrap behavior with empty test sets is explicit and compatible with evaluation helpers.

## Test plan
- uv run python -m pytest tests/test_resampling.py tests/test_evaluation.py
- uv run python -m pytest
- uv run ruff check .
